src/Benchmarks/Middleware/JsonMiddleware.cs

```diff
@@ -53,7 +56,11 @@ public Task Invoke(HttpContext httpContext)
 
                 return Task.CompletedTask;
 #else
-                return JsonSerializer.SerializeAsync<JsonMessage>(httpContext.Response.Body, new JsonMessage { message = "Hello, World!" });
+                _writer ??= new Utf8JsonWriter(httpContext.Response.Body);
+                _writer.Reset(httpContext.Response.Body);
+
+                JsonSerializer.Serialize<JsonMessage>(_writer, new JsonMessage { message = "Hello, World!" }, CustomJsonContext.Default.JsonMessage);
```

Comment:
I thought `JsonSerializer.Serialize` will internally pool the `Utf8JsonWriter` instance it uses if we don't pass one?

Reply:
No https://source.dot.net/#System.Text.Json/System/Text/Json/Serialization/JsonSerializer.Write.Stream.cs,391
And also, the source-gen isn't called when using the stream overloads. https://github.com/dotnet/runtime/pull/78646

Reply:
So which overloads do pool? Just those that return string?

Reply:
> Just those that return string?

Looks like it

Reply:
All serialization overloads pool `Utf8JsonWriter`, with the exception of the async streaming serialization overloads that can't use the fast path.